src/device/pif/pif.c

```diff
@@ -371,6 +372,12 @@ void update_pif_ram(struct pif* pif)
         input.readController(-1, NULL);
     }
 
+    if (netplay_is_init())
```

Comment:
This should go inside netplay input plugin. You may have to extend the input backend API (or use the readController(-1, NULL) trick).

Reply:
I'm not really sure how to do this. update_pif_ram calls ```input.readController(-1, NULL)```. This is the point that a raw input plugin like Raphnet updates the PIF ram. I need to intercept that data before update_pif_ram exits

Reply:
Probably in main_run you can store the value of input.readController, replace it with one you provides which calls the original input.reacController and adds your netplay logic after. Something like:
```
void netplay_read_controller(int control, unsigned char* command)
{
    original_read_controller(control, command);
   
    netplay_send_raw_input(&g_dev.pif);
    netplay_get_raw_input(&g_dev.pif);
}

// maybe inside main_run ? or somewhere after input plugin setup ?
...
if (netplay_is_init()) {
  original_read_controller = input.readController;
  input.readController = netplay_read_controller;
}
```

Reply:
I think at that point (hijacking the call to input.readController), it's getting a little hacky just to remove a few lines of code. I consolidated the functions inside update_pif_ram to a single ```netplay_update_input``` function, so it's just a single line of code, similar to the new_vi call in the vi_controller.